kyuubi-server/src/main/scala/org/apache/kyuubi/server/KyuubiTHttpFrontendService.scala

```diff
@@ -0,0 +1,278 @@
+
+      info("Started " + getClass.getSimpleName +
+        " in " + schemeName + " mode on port " + portNum +
+        " path=" + httpPath + " with " + minThreads + "..." + maxThreads + " worker threads")
+    } catch {
+      case e: Throwable =>
+        MetricsSystem.tracing(_.incCount(THRIFT_CONN_FAIL))
+        error(e)
+        throw new KyuubiException(
+          s"Failed to initialize frontend service on $serverAddr:$portNum.",
+          e)
+    }
+    super.initialize(conf)
+  }
+
+  override def run(): Unit =
+    try {
+      if (isServer()) {
+        info(s"Starting and exposing JDBC connection at: jdbc:hive2://$connectionUrl/")
+      }
+      server.foreach(_.start())
+    } catch {
+      case _: InterruptedException => error(s"$getName is interrupted")
+      case t: Throwable =>
+        error(s"Error starting $getName", t)
+        System.exit(-1)
+    }
+
+  override protected def stopServer(): Unit = {
+    server.foreach(_.stop())
+    server = None
+  }
+
+  override protected def isServer(): Boolean = true
+
+  override val discoveryService: Option[Service] = None
+
+  private def getHttpPath(httpPath: String): String = {
+    if (httpPath == null || httpPath == "") return "/*"
+    else {
+      if (!httpPath.startsWith("/")) return "/" + httpPath
+      if (httpPath.endsWith("/")) return httpPath + "*"
+      if (!httpPath.endsWith("/*")) return httpPath + "/*"
+    }
+    httpPath
+  }
+
+  def constrainHttpMethods(ctxHandler: ServletContextHandler): Unit = {
+    val constraint = new Constraint
+    constraint.setAuthenticate(true)
+    val cmt = new ConstraintMapping
+    cmt.setConstraint(constraint)
+    cmt.setMethod("TRACE")
+    cmt.setPathSpec("/*")
+    val securityHandler = new ConstraintSecurityHandler
+    val cmo = new ConstraintMapping
+    cmo.setConstraint(constraint)
+    cmo.setMethod("OPTIONS")
+    cmo.setPathSpec("/*")
+    securityHandler.setConstraintMappings(Array[ConstraintMapping](cmt, cmo))
+    ctxHandler.setSecurityHandler(securityHandler)
+  }
+
+  override protected def getIpAddress: String = {
+    SessionManager.getIpAddress
+  }
+
+  override protected def getUserName(req: TOpenSessionReq): String = {
+    var userName: String = SessionManager.getUserName
+    if (userName == null) userName = req.getUsername
+    userName = getShortName(userName)
+    val effectiveClientUser: String = getProxyUser(req.getConfiguration, getIpAddress, userName)
+    debug("Client's username: " + effectiveClientUser)
+    effectiveClientUser
+  }
+
+  private def getShortName(userName: String): String = {
+    var ret: String = null
+
+    if (userName != null) {
+      if (authFactory.isKerberosEnabled) {
+        // KerberosName.getShorName can only be used for kerberos user, but not for the user
+        // logged in via other authentications such as LDAP
+        val fullKerberosName = ShimLoader.getHadoopShims.getKerberosNameShim(userName)
+        ret = fullKerberosName.getShortName
```

Comment:
question: kyuubi support enable kerberos and LDAP in same time, is it sufficient to handle that case?